src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java

```diff
@@ -412,4 +415,93 @@ void shouldReturn500ForStatusOnTriggeredFailure() {
+                    .path("/eureka/v2/apps/{appId}")
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertNoContentResponse(response);
+            verify(eurekaServer).registerApplication(isA(InstanceInfo.class));
+        }
+
+        @Test
+        void shouldAttemptToRegisterAndReturn500OnTriggeredFailure() {
+            var postBody = Map.of(
+                    "instance", Map.of(
+                            "hostName", "localhost",
+                            "homePageUrl", "http://localhost",
+                            "statusPageUrl", "http://localhost/status",
+                            "healthCheckUrl", "http://localhost/healthcheck",
+                            "status", "STARTING",
+                            "port", Map.of("$", 8080),
+                            "securePort", Map.of("$", 8081),
+                            "vipAddress", "RegisterUseResponseStatusCode-500"
+                    )
+            );
+
+            var response = client.target(server.getURI())
+                    .path("/eureka/v2/apps/{appId}")
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertInternalServerErrorResponse(response);
+            verifyNoInteractions(eurekaServer);
+        }
+
+        @Test
+        void shouldAttemptToRegisterAndReturn500OnTriggeredFailure_ThenSuccessAfterTriggeredNumber() {
+            var postBody = Map.of(
+                    "instance", Map.of(
+                            "hostName", "localhost",
+                            "homePageUrl", "http://localhost",
+                            "statusPageUrl", "http://localhost/status",
+                            "healthCheckUrl", "http://localhost/healthcheck",
+                            "status", "STARTING",
+                            "port", Map.of("$", 8080),
+                            "securePort", Map.of("$", 8081),
+                            "vipAddress", "FailRegistrationFirstNTimes-2"
+                    )
+            );
+
+            for (var i = 0; i < 2; i++) {
+                var response = client.target(server.getURI())
+                        .path("/eureka/v2/apps/{appId}")
+                        .resolveTemplate("appId", APP_NAME)
+                        .request(MediaType.APPLICATION_JSON_TYPE)
+                        .post(json(postBody));
+
+                assertInternalServerErrorResponse(response);
+            }
+
+            var response = client.target(server.getURI())
+                    .path("/eureka/v2/apps/{appId}")
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertNoContentResponse(response);
```

Comment:
OK so I assume this is testing what I asked about earlier? i.e. that it fails up to a specific number of retries, and then finally returns success???

Reply:
Correct